include/s3select_csv_parser.h

```diff
@@ -1,413 +1,255 @@
+                                - std::begin(column_names));
+                }
+
+                void set_file_name(const std::string&file_name){
+                        in.set_file_name(file_name);
+                }
+
+                void set_file_name(const char*file_name){
+                        in.set_file_name(file_name);
+                }
+
+                const char*get_truncated_file_name()const{
+                        return in.get_truncated_file_name();
+                }
+
+                void set_file_line(unsigned file_line){
+                        in.set_file_line(file_line);
+                }
+
+                unsigned get_file_line()const{
+                        return in.get_file_line();
+                }
+
+        private:
+                void parse_helper(std::size_t){}
+
+                //template<class T, class ...ColType>
+                void parse_helper(std::size_t r, std::vector<char*>& cols){
+                        char* t;
+                        if(row[r] && r<=(column_count-1)){
+                                try{
+                                        try{
+                                                ::io::detail::parse<overflow_policy>(row[r], t);
+                                                cols.push_back(t);
+                                        }catch(error::with_column_content&err){
+                                                err.set_column_content(row[r]);
+                                                throw;
+                                        }
+                                }catch(error::with_column_name&err){
+                                        err.set_column_name(column_names[r].c_str());
+                                        throw;
+                                }
+                                parse_helper(r+1, cols);
+                        }
+                }
+
+
+        public:
+                //template<class ...ColType>
+                bool read_row(std::vector<char*>& cols){
+                        /*static_assert(sizeof...(ColType)>=column_count,
+                                "not enough columns specified");
+                        static_assert(sizeof...(ColType)<=column_count,
+                                "too many columns specified");*/
+                        try{
+                                try{
+
+                                        char*line;
+                                        do{
+                                                line = in.next_line();
+                                                if(!line)
+                                                        return false;
+                                        }while(comment_policy::is_comment(line));
+
+                                        detail::parse_line<trim_policy, quote_policy>
+                                                (line, row, col_order);
+
+                                        parse_helper(0, cols);
```

Comment:
I think we can skip the `parse_helper`, it checks column-types 
our engine is NOT using column-type in the case of CSV(all columns are string)